examples/src/main/scala/com/gu/tableversions/examples/PageviewLoader.scala

```diff
@@ -0,0 +1,60 @@
+package com.gu.tableversions.examples
+
+import java.sql.{Date, Timestamp}
+import java.time.{ZoneId, ZonedDateTime}
+
+import com.gu.tableversions.examples.PageviewLoader.Pageview
+import org.apache.spark.sql.{Dataset, SaveMode, SparkSession}
+
+/**
+  * This example contains code that writes example event data to a table that has a single date partition.
+  * It demonstrates how individual partitions in such a table can be updated using versioning.
+  *
+  * @param tableName the fully qualified table name that will be populated by this loader
+  * @param tableUri The location where the table data will be stored
+  */
+class PageviewLoader(tableName: String, tableUri: String)(implicit val spark: SparkSession) {
+
+  import spark.implicits._
+
+  def initTable(): Unit = {
+    val ddl = s"""CREATE EXTERNAL TABLE IF NOT EXISTS $tableName (
+                 |  `id` string,
+                 |  `path` string,
+                 |  `timestamp` timestamp
+                 |)
+                 |PARTITIONED BY (`date` date)
+                 |STORED AS parquet
+                 |LOCATION '$tableUri'
+    """.stripMargin
+
+    spark.sql(ddl)
+    ()
+  }
+
+  def pageviews(): Dataset[Pageview] =
+    spark.table(tableName).as[Pageview]
+
+  def insert(dataset: Dataset[Pageview])(implicit spark: SparkSession): Unit = {
```

Comment:
Can we use the `SparkSession` from the constructor? Alternatively, can we pass the `SparkSession` into each method that requires it (implicitly or otherwise)

Reply:
Ah yes, it should use the constructor one, that's a not-quite-complete refactoring.